### PR TITLE
Database update and authentication

### PR DESCRIPTION
**Pull request by FiIipson**
It's ready to merge since contains @MajkeIos datebase update

### CODE REVIEW

File: src/main/java/com/example/learninghub/authentication/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/auth")
```
Comment (m-sowinski) [inline src/main/java/com/example/learninghub/authentication/AuthenticationController.java]: Change it to match /v1/* convention

File: src/main/java/com/example/learninghub/authentication/DemoController.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.example.learninghub.authentication;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/demo-controller")
```
Comment (m-sowinski) [inline src/main/java/com/example/learninghub/authentication/DemoController.java]: same here

File: src/main/java/com/example/learninghub/authentication/SecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private final JwtAuthenticationFilter jwtAuthFilter;
+    private final AuthenticationProvider authenticationProvider;
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity
+                .csrf()
+                .disable()
+                .authorizeHttpRequests()
+                .requestMatchers("/api/v1/auth/**")
```
Comment (m-sowinski) [inline src/main/java/com/example/learninghub/authentication/SecurityConfiguration.java]: also here

File: src/main/resources/setup/sample_data.sql [hunk 4]
```diff
@@ -1,14 +1,48 @@
...
+    (1, 'Sample input 1', 'Sample output 1'),
+    (2, 'Sample input 2', 'Sample output 2'),
+    (3, 'Sample input 3', 'Sample output 3'),
+    (4, 'Sample input 4', 'Sample output 4'),
+    (5, 'Sample input 5', 'Sample output 5');
+
+INSERT INTO
+    _user(username, password, email, role)
+VALUES
+    ('user 1', 'password 1', 'user1@email.com', 'standard');
```
Comment (m-sowinski) [inline src/main/resources/setup/sample_data.sql]: I think the whole 'standard' word must be in capital letters, but if it works fine, skip this comment

**Review by m-sowinski — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/learninghub/authentication/AuthenticationController.java (hunk 1 above)
Comment (FiIipson) [reply src/main/java/com/example/learninghub/authentication/AuthenticationController.java]: Done

File: src/main/java/com/example/learninghub/authentication/DemoController.java (hunk 2 above)
Comment (FiIipson) [reply src/main/java/com/example/learninghub/authentication/DemoController.java]: Done

File: src/main/java/com/example/learninghub/authentication/SecurityConfiguration.java (hunk 3 above)
Comment (FiIipson) [reply src/main/java/com/example/learninghub/authentication/SecurityConfiguration.java]: Done

File: src/main/resources/setup/sample_data.sql (hunk 4 above)
Comment (FiIipson) [reply src/main/resources/setup/sample_data.sql]: Done

**Pull request merged**